compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -279,7 +280,26 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     hir::ExprKind::Break(self.lower_jump_destination(e.id, *opt_label), opt_expr)
                 }
                 ExprKind::Continue(opt_label) => {
-                    hir::ExprKind::Continue(self.lower_jump_destination(e.id, *opt_label))
+                    if let Some(_) = opt_label {
+                        if let Some((_, is_loop, block_span)) = self.resolver.get_label_res(e.id) {
+                            if is_loop {
```

Comment:
Could you use a let-chain for the error condition, and only call the good path once?
I mean:
```
if opt_label.is_some() && let Some((_, false, block_span)) = self.resolver.get_label_res(e.id) {
    ExprKind::Err(...)
} else {
    ExprKind::Continue(...)
}
```